addons/spreadsheet/static/src/data_sources/display_name_repository.js

```diff
@@ -0,0 +1,200 @@
+/** @odoo-module */
+
+import { Deferred } from "@web/core/utils/concurrency";
+import { LoadingDataError } from "../o_spreadsheet/errors";
+import BatchEndpoint, { Request } from "./server_data";
+
+/**
+ * @typedef PendingDisplayName
+ * @property {"PENDING"} state
+ * @property {Deferred<string|undefined>} deferred
+ *
+ * @typedef ErrorDisplayName
+ * @property {"ERROR"} state
+ * @property {Deferred<string|undefined>} deferred
+ * @property {Error} error
+ *
+ * @typedef CompletedDisplayName
+ * @property {"COMPLETED"} state
+ * @property {Deferred<string|undefined>} deferred
+ * @property {string|undefined} value
+ *
+ * @typedef {PendingDisplayName | ErrorDisplayName | CompletedDisplayName} DisplayNameResult
+ *
+ * @typedef {[number, string]} NameGetRPCResult
```

Comment:
The average dev stumbling on this would expect an array of arrays.
or maybe name it `BatchedNameGetRPCResult` ?
I don't know